### ISSUE TITLE
minishift profile set xxxx reports “…: no such file or directory”

### ISSUE THREAD

**Issue by amitkrout** [minishift/minishift#2666]
### General information

  * Minishift version: v1.22.0/master
  * OS: Linux / macOS / Windows
  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

### Steps to reproduce

  1. Make sure no previous minishift configuration should present on the system
  2. Make sure no minishift sub command should run which creates the ```.minishift``` dir
  3. 
```
$ ./minishift profile set Test
F0805 23:18:08.339053    1653 root.go:389] Cannot create file '/Users/amit/.minishift/config/global.json': open /Users/amit/.minishift/config/global.json: no such file or directory
```
  4. 

### Expected
Should create ```Test``` profile. However running the same command after profile list set the ```Test``` profile.
```
$ ./minishift profile set Test
F0805 23:18:08.339053    1653 root.go:389] Cannot create file '/Users/amit/.minishift/config/global.json': open /Users/amit/.minishift/config/global.json: no such file or directory
$ ./minishift profile list
- Test		Does Not Exist
- minishift	Does Not Exist	(Active)
$ ./minishift profile set Test
Profile 'Test' set as active profile.
```
### Actual
Creates the ```Test``` profile despite of the weird output, but it does not set it as active profile 
 ```
$ ./minishift profile set Test
F0805 23:18:08.339053    1653 root.go:389] Cannot create file '/Users/amit/.minishift/config/global.json': open /Users/amit/.minishift/config/global.json: no such file or directory
 $ ./minishift profile list
- Test		Does Not Exist
- minishift	Does Not Exist	(Active)
```
### Logs

You can start Minishift with `minishift start --show-libmachine-logs -v5` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by LalatenduMohanty** [comment]
@amitkrout @anjannath I can not reproduce this issue on mac. Is it a windows specific issue?

**Comment by LalatenduMohanty** [comment]
It is reproducible when ~/.minishift directory is not present.

``` 
$ mv .minishift .minishift_backup
$ minishift profile set Test
F0806 16:55:03.031042    3686 root.go:389] Cannot create file '/Users/lmohanty/.minishift/config/global.json': open /Users/lmohanty/.minishift/config/global.json: no such file or directory
```

**Comment by 123avi** [comment]
Getting the same 
```  ./minishift profile set tutorial ```
result : 
``` Cannot create file '/home/avi/minishift-1.22.0/config/global.json': open /home/avi/minishift-1.22.0/config/global.json: no such file or directory ```

**Comment by LalatenduMohanty** [comment]
Resolved via #2670
